// telegram-framework/dist/Response.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const Utils_1 = require("./Utils");
const Keyboard_1 = require("./Keyboard");
class Response {
    constructor(user, bot, storage, query) {
        this.user = user;
        this.bot = bot;
        this.storage = storage;
        this.query = query;
    }
    reply(msg, keyboard) {
        this.bot.sendMessage(this.query.chat.id, msg, Object.assign(Object.assign({}, keyboard.build()), { reply_to_message_id: this.query.message_id }));
    }
    message(msg, options) {
        if (options instanceof Keyboard_1.default)
            options = options.build();
        return this.bot.sendMessage(this.query.chat.id, msg, options);
    }
    redirect(path) {
        this.user.history.push(this.user.path);
        this.user.path = (0, Utils_1.parseURL)(path);
        this.storage.Set(this.user);
        this.bot.emit("message", Object.assign(Object.assign({}, this.query), { text: null }));
    }
    push(path) {
        this.user.history.push(this.user.path);
        this.user.path.push(...(0, Utils_1.parseURL)(path));
        this.storage.Set(this.user);
        this.bot.emit("message", Object.assign(Object.assign({}, this.query), { text: null }));
    }
    pushBack() {
        this.user.path = this.user.history.pop();
        this.storage.Set(this.user);
        this.bot.emit("message", Object.assign(Object.assign({}, this.query), { text: null }));
    }
}
exports.default = Response;
